### DISCUSSION TITLE
Hot reload

### DISCUSSION THREAD

**Discussion by Arvant** [axmolengine/axmol#1165] [General]
how any whey two develop axmol game c++ with hot reload. i think it can help develop speed. i try to use visual studio c++ hot reload but not work for me.

**Comment by aismann** [comment]
Maybe this is helping?

https://programmingcsharp.com/enable-hot-reload-in-visual-studio/

Use at least Visual Studio 2022  and .NET 6
Also, you need to use .NET 6 as a framework.

Anyhow its more a visual studio option not realy depend on axmol

**Reply by DelinWorks** [reply]
The last time i checked It doesn't work for me too with axmol but works for other C++ projects that don't use axmol, it might be something to do with the cmake configuration I don't know really..

**Comment by rh101** [comment]
It may not work with Axmol, and it has something to do with the PDB generation.  The file gets too large, and you start getting errors when using any debug database format other than `C7 compatible (/Z7)`.  By default all libraries, including axmol, are all set to `/Zi`.

If you want to try it out, then just add this to your `CMakeLists.txt`:

    target_compile_options(${APP_NAME}
        PUBLIC "$<$<CXX_COMPILER_ID:MSVC>:/ZI>"
    )

The `/ZI` flag is what creates the correct debug information format for hot reload, as you can see here:

![image](https://user-images.githubusercontent.com/8603230/229983513-61d1c5dc-7432-4400-a956-39e4cd0ed457.png)

It will most likely show errors regarding the PDB size, so if that is the case, then `/Z7` is your only option, which does not work with hot reload.
